web/Auth/ClaimsTransformer.cs

```diff
@@ -51,8 +51,11 @@ public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
         var claims = new List<Claim>(existingClaimsIdentity.Claims);
         // If they're an admin in the database, add the role to their claims
         if (databaseUser.IsAdmin) claims.Add(new Claim(ClaimTypes.Role, "Admin"));
-        if (databaseUser.LandlordId != null) claims.Add(new Claim(ClaimTypes.Role, "Landlord"));
-
+        if (databaseUser.LandlordId != null)
+        {
+            var userLandlord = _context.Landlords.FirstOrDefault(l => l.Id == databaseUser.LandlordId);

```

Comment:
So instead of this,we could just grab the landlord at the first call, rather than call again. On line 38 we can add a `.Include(u => u.Landlord)`. Also is there a reason we are using first rather than Single?

If we do it this way then we can make the check into
```
        if (databaseUser.Landlord is { Disabled: false })
        {
            claims.Add(new Claim(ClaimTypes.Role, "Landlord"));
        }
```

Reply:
Ah, I tried grabbing the landlord at first call, but without the .Include(u=> u.Landlord) it was (inexplicably, as far as I could tell) returning null - it's much nicer getting the Landlord directly.
As to why it's First rather than Single - there might have been a reason I did that, or it might have just been that I was tired at the time and not thinking properly - I'll change it.